h2/src/main/org/h2/command/Parser.java

```diff
@@ -8013,6 +8013,15 @@ private void parseTableColumnDefinition(CommandWithColumns command, Schema schem
             if (NullConstraintType.NULL_IS_NOT_ALLOWED == parseNotNullConstraint()) {
                 column.setNullable(false);
             }
+            String comment = readCommentIf();
+            if (comment != null) {
+                if (column.getComment() != null) {
+                    addExpected(CHECK);
+                    addExpected("REFERENCES");
+                    throw getSyntaxError();
```

Comment:
These three lines throw incomplete exception for duplicate comment.

But with swapped conditions they should be simply removed.